### PR TITLE
Feat/support for expressions in group label

### LINKED ISSUE #1606: Dynamic expressions not supported in code lists based on repeating groups
### Description

When trying to provide a dynamic expression to the values `source.label`, `source.helpText` and `source.description`, the application crashes. I would expect to be able to run dynamic expressions in these properties. 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by mikaelrss**
## Description

Adds support for dynamic expressions in `label`, `description`, and `helpText` of the `source` property for code lists based on repeating groups. 

## Related Issue(s)

- closes #1606

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by mikaelrss** [comment]
Percy is failing with 22 screenshots because I added two new fields for the cypress tests in the `frontend-test` app. No errors.

**Comment by mikaelrss** [comment]
The code is covered by Cypress tests, so I think we can ignore the coverage failure from SonarCloud

**Review by olemartinorg — CHANGES_REQUESTED** [review]

File: src/features/expressions/index.ts [hunk 1]
```diff
@@ -245,6 +245,13 @@ export function argTypeAt(func: ExprFunction, argIndex: number): ExprVal | undef
   return undefined;
 }
 
+export function isExpression(input: string | Expression | undefined): input is Expression {
+  if (!Array.isArray(input)) {
+    return false;
+  }
+  return validateRecursively(input, { errors: {} }, []) !== undefined;
+}
```
[the remark this answers is not in the document]
Comment (mikaelrss) [reply src/features/expressions/index.ts:253]: Aha, I can change the code to use `asExpression` then

File: src/utils/memoize.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    const key = JSON.stringify(args);
+    if (_cache.has(key)) {
+      return _cache.get(key);
+    }
+
+    const result = fn(...args);
+    _cache.set(key, result);
+    return result;
+  } as T;
+}
```
Comment (olemartinorg) [inline src/utils/memoize.ts:13]: We have a similar utility in `useMemoDeepEqual()`, but I just recently discovered it wasn't working quite like one would expect it to... I have a fix for that underway in `chore/instantation-to-hook` (which is growing quite large now).

Comment (mikaelrss) [reply src/utils/memoize.ts:13]: Ah yes, but this is hook depeendent, right? The `memoize` function is called outside a react-hook, that's why I made it :partying_face:

**Review by olemartinorg — APPROVED** [review]

**Comment by olemartinorg** [comment]
Tested, and it's working great! 🥳 

But I don't quite agree with this statement.. 😉 
> - User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
>   - [ ] Has been added/updated
>   - [x] No functionality has been changed/added, so no documentation is needed
>   - [ ] I will do that later/have created an issue
>   

The documentation [located here](https://docs.altinn.studio/nb/app/development/logic/expressions/#bruksomr%C3%A5der) no longer reflects the actual features available in the frontend. I believe you can simply just add this to the [app-frontend-v4](https://github.com/Altinn/altinn-studio-docs/tree/app-frontend-v4) branch where we'll document stuff that arrives in v4 - no PR needed.

**Comment by mikaelrss** [comment]
> Tested, and it's working great! 🥳
> 
> But I don't quite agree with this statement.. 😉
> 
> > * User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
> >   
> >   * [ ]  Has been added/updated
> >   * [x]  No functionality has been changed/added, so no documentation is needed
> >   * [ ]  I will do that later/have created an issue
> 
> The documentation [located here](https://docs.altinn.studio/nb/app/development/logic/expressions/#bruksomr%C3%A5der) no longer reflects the actual features available in the frontend. I believe you can simply just add this to the [app-frontend-v4](https://github.com/Altinn/altinn-studio-docs/tree/app-frontend-v4) branch where we'll document stuff that arrives in v4 - no PR needed.

Whooops! Alright. I'll definitely add the documentation to the v4 branch :smile:

**Pull request merged**